### DISCUSSION TITLE
New Logo and Icon Proposal

### DISCUSSION THREAD

**Discussion by whatsnewsisyphus** [iina/iina#3631] [Ideas]
Hello All, as a longtime iina user, I was wondering what people thought about a new logo and icon that is more consistent with the visual language of the interface. It will also bring some coherency across the board. This is just a quick sketch of the concept, we can talk and workshop the particulars such as colors, line thickness, sharpness, angles, and so forth
![iina-01](https://user-images.githubusercontent.com/454772/152279745-33ba0f5d-2a07-4b29-8daf-88f0f8c76735.png)
![iina-02](https://user-images.githubusercontent.com/454772/152279744-d524b3a9-055a-4570-876e-464683725217.png)
![iina-03](https://user-images.githubusercontent.com/454772/152279743-8a0620a1-b616-4478-bfb6-39cdfcc9ac98.png)

**Comment by setanarut** [comment]
i like second icon but i prefer darkmode version. (dark background)

**Comment by lemikeone** [comment]
I really like it too, great job! I agree with @hazarek, it would be great to have a dark version. 

Unfortunately, I'm not sure that IINA is still maintained, as the latest version is now one year old (but it's still my default media player). IINA 1.2.0 was released on 6 Feb 2021.

**Comment by whatsnewsisyphus** [comment]
@hazarek here's a dark preview. It's a pity that IINA is not being developed, hope somebody picks it up sometime. I'm still happy to make the assets if people care to rebuild
![iina-04](https://user-images.githubusercontent.com/454772/152373626-a56b7c09-ad68-44e9-bfc7-ff715f182a50.png)
![iina-05](https://user-images.githubusercontent.com/454772/152373643-8ce92a38-27b9-4b6d-91f9-b25f7311bd2c.png)
.

**Comment by low-batt** [comment]
Not abandoned! Goto "Pull requests". Check the dates of the PRs. You will see that there has been slow steady continuous software development efforts since June last year to investigate and propose fixes for outstanding defects.

A new release is currently under development. The development team is working through the large backlog of proposed fixes. Work is proceeding steadily ahead at maximum galloping turtle speed as per software development norms.

On the subject at hand. Wow. Really cool looking. I'm very impressed. I will mention this to the development team, but at the moment I think they need to stay focused on getting a bug fix release out the door.

**Reply by lemikeone** [reply]
That's great to hear! Thanks for your work.

**Comment by whatsnewsisyphus** [comment]
@low-batt That is wonderful news. Just let me know and I can make the assets to OSX specs and make a pull request. Some color trials below.
![iina-05](https://user-images.githubusercontent.com/454772/154889834-df0fd9e7-f89c-4a27-a39e-72a20b0e4025.png)
![iina-04](https://user-images.githubusercontent.com/454772/154889835-d6455c6e-990b-448c-a5b1-9c062069658b.png)
![iina-03](https://user-images.githubusercontent.com/454772/154889837-11db4ab6-6477-4164-8557-483cb3111c5b.png)
![iina-02](https://user-images.githubusercontent.com/454772/154889838-bd7f5a7c-40cc-422b-9047-7a63e4e19601.png)
![iina-01](https://user-images.githubusercontent.com/454772/154889839-903eef42-377b-4926-9957-6239dfd6dd91.png)

**Comment by low-batt** [comment]
Officially I am not a member of the IINA development group. I am an IINA user that heard the developers were overloaded and decided to help out by investigating problems and submitting proposed code changes to address them. I am currently working with the developers as they are reviewing the PRs I submitted. So anything I say here is **unofficial** and just *my own opinions*.

What is in play here is some of the common issues in making a proposal to a group of people. The more work you put into a proposal the more likely that group of people will not reject your proposal because they don't understand your vision. Of course the more work you put into a proposal the more of your time and effort you risk seeing thrown away if your proposal is rejected. Attempting to mitigate this risk by asking up front if that group would be open to your proposal comes with the risk that the group says no because they don't understand the proposal.

The way this applies to this discussion is whether or not you take this all the way to a pull request that contains the proposed changes. Officially IINA does have an opinion on this. From [CONTRIBUTING](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md):
> First of all, if you plan on doing any work, please check the issue tracker. Not only does this make it easier to track progress, it also prevents people doing duplicate work, as well as allows for the community to weigh in on the implementation of a feature before it's already all coded up. Nothing's more sad than having to go back and delete or rewrite code because of a misunderstanding in how it would fit into IINA :(

So the IINA project would have you ask up front before working on changes. Personally I'm very glad that IINA is concerned about not wasting the time of potential contributors. But in my case I did not ask up front since I knew the developers were busy. I just accepted the risk that I might need to do rework or if I really got it wrong my submission might end up in the trash.

Another common issue with making a proposal is **timing**. Ignore for the moment that the development team is very late on getting out a bug fix release and needs to be nose to the grindstone fully focused on getting that release out. Due to users starting to assume the project is dead, my own opinion is that this next release should not contain such "branding" changes. Because it has been so long since there was an IINA release you want the released application to have the same "look" to assure users this is the IINA they are used to. There is a release looming that would be more appropriate for this kind of change. A plug-in system is under development. I don't know all the details. Some of what it addresses is the current issues with embedding youtube-dl and code for downloading subtitles from sites. Because that will be a significant feature change release, that is the kind of update users might expect visual changes in.

Remember all of this is just my opinions. I don't speak for the development team. I have no idea how amenable to an icon change the team might be.

As for how to go about changing assets. I'm not a graphical user interface programmer. I have no clue. I've mostly been leaving UI issues to others to solve.

**Reply by whatsnewsisyphus** [reply]
Ok, I'll hold on it for a bit to let them release the fix release and make an issue tracker where the community can have their input on the design language

**Reply by whatsnewsisyphus** [reply]
I appreciate the detailed response.

**Comment by sunlit777** [comment]
yeah, color management is broken and we are discussing new LOGOS.

**Comment by low-batt** [comment]
It is very typical of software projects to concurrently work on multiple issues and not have the entire team focused solely on one top priority problem.

That said, in the case of this discussion it has indeed been sidelined due to the development team being totally overloaded and not having time to consider such changes.

When it comes to color management, that is mostly handled by the library from the [mpv project](https://github.com/mpv-player/mpv) that IINA uses for playing videos. There are known mpv problems with color management. Look for mpv issues labeled with `core:color-management`. If there is a color issue where mpv works fine and IINA does not then that need to be looked into. At the moment the IINA team is focused on preparing a release with upgraded libraries from both mpv and FFmpeg as well as other dependencies.

**Comment by sunlit777** [comment]
but player is unusable without working color management on any modern mac! think about it! Sent from my iPhoneOn 23 Jul 2024, at 04:56, low-batt ***@***.***> wrote:﻿
It is very typical of software projects to concurrently work on multiple issues and not have the entire team focused solely on one top priority problem.
That said, in the case of this discussion it has indeed been sidelined due to the development team being totally overloaded and not having time to consider such changes.
When it comes to color management, that is mostly handled by the library from the mpv project that IINA uses for playing videos. There are known mpv problems with color management. Look for mpv issues labeled with core:color-management. If there is a color issue where mpv works fine and IINA does not then that need to be looked into. At the moment the IINA team is focused on preparing a release with upgraded libraries from both mpv and FFmpeg as well as other dependencies.

**Comment by low-batt** [comment]
This is off topic. Please enter an [issue](https://github.com/iina/iina/issues) for the color management problem.
